haskell-backend/src/main/java/org/kframework/backend/haskell/HaskellBackend.java

```diff
@@ -20,9 +22,10 @@
     @Inject
     public HaskellBackend(
             KompileOptions kompileOptions,
+            KProveOptions kproveOptions,
```

Comment:
Architecturally, this is wrong. It should not require a KProveOptions object to construct something unless you are within the kprove tool's code path, but backends can also be constructed by kompile

Reply:
Yes I'm aware of this. Actually there are other changes needed that will make access to KProveOptions unnecessary.